### PR TITLE
feat(no-unused-refs): support template ref api

### PR DESCRIPTION
**Pull request by KazariEX**

### CODE REVIEW

**Comment by FloEdelmann** [comment]
Thanks! But can you please elaborate on the change? Where is `useTemplateRef` documented? Is it a native Vue feature?

**Comment by KazariEX** [comment]
It is a new feature of Vue 3.5 and there is no documentation yet.

https://github.com/vuejs/core/blob/main/packages/runtime-core/src/helpers/useTemplateRef.ts

File: lib/rules/no-unused-refs.js [hunk 1]
```diff
@@ -233,6 +233,26 @@ module.exports = {
               refsNode = id.parent
             }
             extractUsedForPattern(refsNode)
+          },
+          CallExpression(exp) {
+            if (
+              exp.callee.name !== 'useTemplateRef' &&
+              exp.callee.name !== 'templateRef'
```
Comment (ota-meshi) [inline lib/rules/no-unused-refs.js:240]: I don't think `templateRef` will be provided in Vue 3.5 either. Could you please remove `templateRef` check?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: lib/rules/no-unused-refs.js (hunk 1 above)
Comment (KazariEX) [reply lib/rules/no-unused-refs.js:240]: Due to the alias support provided by language core for templateRef, I have habitually added it. Now remove it :)

File: lib/rules/no-unused-refs.js [hunk 2]
```diff
@@ -233,6 +233,23 @@ module.exports = {
...
+          CallExpression(exp) {
+            if (exp.callee.name !== 'useTemplateRef') {
+              return
+            }
+            const arg0 = exp.arguments[0]
+            if (!arg0) {
+              return
+            }
+            if (arg0.type === 'Literal') {
+              usedRefs.add(arg0.value)
```
Comment (ota-meshi) [inline lib/rules/no-unused-refs.js:246]: Could you please use `utils.getStringLiteralValue()`? Please refer to the code of custom-event-name-casing for usage.

https://github.com/vuejs/eslint-plugin-vue/blob/d77fbf7a05f3a95085ddbfeec3eaee576e35ecd7/lib/rules/custom-event-name-casing.js#L32

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Review by FloEdelmann — APPROVED** [review]
Fine from my side :slightly_smiling_face:

**Comment by FloEdelmann** [comment]
FYI: Vue 3.5 with `useTemplateRef` was released today: https://blog.vuejs.org/posts/vue-3-5#usetemplateref

**Comment by s3xysteak** [comment]
Now it is the most recommended way to use template ref https://vuejs.org/guide/essentials/template-refs.html#accessing-the-refs. Hope it can be merged soon 😄 .

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**